runtime/compiler/env/VMJ9.cpp

```diff
@@ -7488,120 +7488,110 @@ TR_J9VM::inlineNativeCall(TR::Compilation * comp, TR::TreeTop * callNodeTreeTop,
+         //
+         // Depth  Caller Index  Method
+         //   [3]             2  getCallerClass()
+         //   [2]             1  Method.invoke()
+         //   [1]             0  Method.invoke()
+         //   [0]            -1  Method.invoke()
+         //
+         // We cannot do this reduction in such a case because we want to get the class of the non-skipped caller
+         // of the call at depth [0] but depths [0] is the top level method we are compiling.
+         bool skipFrameAtDepth0 = false;
 
-            bool skipFrame;  // the logic of computing skipFrame should match with what vm does
-            while (true)
+         while (true)
+            {
+            if (callerIndex == -1)
                {
-               if (callerIndex != -1)
-                  {
-                  callerMethod = (J9Method *) comp->getInlinedCallSite(callerIndex)._methodInfo;
-                  callerClass = (J9Class*) comp->getInlinedResolvedMethod(callerIndex)->containingClass();
-                  }
-               else
-                  {
-                  TR::ResolvedMethodSymbol *callerMethodSymbol = comp->getJittedMethodSymbol();
-                  callerMethod = (J9Method *) callerMethodSymbol->getResolvedMethod()->getPersistentIdentifier();
-                  callerClass = (J9Class*) callerMethodSymbol->getResolvedMethod()->classOfMethod();
-                  }
-
-               skipFrame = transformJlrMethodInvoke(callerMethod, callerClass);
-
-               if (!skipFrame && inlineDepth == stackDepth)
-                  break;
-               if (callerIndex == -1)
-                  break;
-               if (!skipFrame)
-                  inlineDepth++;
-               callerIndex = comp->getInlinedCallSite(callerIndex)._byteCodeInfo.getCallerIndex();
+               TR::ResolvedMethodSymbol *callerMethodSymbol = comp->getJittedMethodSymbol();
+               callerMethod = reinterpret_cast<J9Method *>(callerMethodSymbol->getResolvedMethod()->getPersistentIdentifier());
+               callerClass = reinterpret_cast<J9Class *>(callerMethodSymbol->getResolvedMethod()->classOfMethod());
                }
-
-            if (!skipFrame && (inlineDepth == stackDepth))
+            else
                {
-               bool returnClassLoader;
-               switch (methodID)
-                  {
-                  case TR::sun_reflect_Reflection_getCallerClass:
-                  case TR::java_lang_invoke_MethodHandles_getStackClass:
-                  //case TR::com_ibm_oti_vm_VM_getStackClass:
-                  //case TR::java_lang_Class_getStackClass:
-                     returnClassLoader = false;
-                     break;
-                  default:
-                     returnClassLoader = true;
-                     break;
-                  }
+               callerMethod = reinterpret_cast<J9Method *>(comp->getInlinedCallSite(callerIndex)._methodInfo);
```

Comment:
Is the inline table entry pointin at a ROM method under AOT ?

Reply:
This optimization is disabled for AOT when not using SVM. Does SVM handle this correctly? I'm not familiar enough to be able to tell. See the start of the implementation:

```
         // We need to bail out since we create a class pointer const with cpIndex of -1
         if (isAOT_DEPRECATED_DO_NOT_USE() && !comp->getOption(TR_UseSymbolValidationManager))
            {
            return 0;
            }
```

Reply:
Okay yes I saw that but my question was around AOT with SVM enabled. @dsouzai can you please let us know of your thoughts on this ?

Reply:
The inlinedCallSite table in AOT is the same as in JIT; we just don't copy it into the J9JITExceptionTable at the end of the compilation because that happens during relocation in an AOT load. As for the `// We need to bail out since we create a class pointer const with cpIndex of -1` in SVM we don't need to worry about cpIndex -1 issues. I don't know exactly where the symref is being created, but we end up just using the original query that was used to get the class as the means to materialize the class during relocation.